src/jtd/converter.ts

```diff
@@ -0,0 +1,248 @@
+ */
+export function convertObjectType(type: ObjectType): jtd.JtdPropertiesForm {
+  const fields = (type as any).fields;
+  const form: jtd.JtdPropertiesForm = {};
+  
+  if (fields && fields.length > 0) {
+    form.properties = {};
+    form.optionalProperties = {};
+    
+    for (const field of fields) {
+      const fieldName = field.key;
+      const fieldType = field.value;
+      
+      if (fieldType) {
+        const fieldJtd = toJtdForm(fieldType);
+        // Check if field is optional using instanceof check since optional is a property
+        if ((field as any).optional === true) {
+          form.optionalProperties[fieldName] = fieldJtd;
+        } else {
+          form.properties[fieldName] = fieldJtd;
+        }
+      }
+    }
+  }
+  
+  // Handle additional properties - check the schema for unknownFields
+  const schema = type.getSchema();
+  if (schema.unknownFields === false) {
+    form.additionalProperties = false;
+  }
+  
+  return form;
+}
+
+/**
+ * JTD form converter for TupleType.
+ */
+export function convertTupleType(type: TupleType<any>): jtd.JtdElementsForm {
+  const elements = (type as any).types;
+  return {
+    elements: elements.map((element: any) => toJtdForm(element))
+  };
+}
+
+/**
+ * JTD form converter for MapType.
+ */
+export function convertMapType(type: MapType<any>): jtd.JtdValuesForm {
+  const valueType = (type as any).type;
+  return {
+    values: toJtdForm(valueType)
+  };
+}
+
+/**
+ * JTD form converter for RefType.
+ */
+export function convertRefType(type: RefType<any>): jtd.JtdRefForm {
+  const ref = type.getRef();
+  return {
+    ref: ref
+  };
+}
+
+/**
+ * JTD form converter for OrType.
+ */
+export function convertOrType(type: OrType<any>): jtd.JtdForm {
+  // OrType is complex - for simplicity, convert to empty form
+  // In a full implementation, this might need to be a discriminator form
+  return convertAbstractType(type);
+}
+
+/**
+ * JTD form converter for BinaryType.
+ */
+export function convertBinaryType(type: BinaryType<any>): jtd.JtdForm {
+  // Binary types don't have a direct JTD equivalent, fall back to default
+  return convertAbstractType(type);
+}
+
+/**
+ * JTD form converter for FunctionType.
+ */
+export function convertFunctionType(type: FunctionType<any, any>): jtd.JtdForm {
+  // Function types don't have a direct JTD equivalent, fall back to default
+  return convertAbstractType(type);
+}
+
+/**
+ * JTD form converter for FunctionStreamingType.
+ */
+export function convertFunctionStreamingType(type: FunctionStreamingType<any, any>): jtd.JtdForm {
+  // Streaming function types don't have a direct JTD equivalent, fall back to default
+  return convertAbstractType(type);
+}
+
+/**
+ * Main router function that converts any AbstractType to JTD form.
+ * Uses a switch statement to route to the appropriate converter function.
+ */
+export function toJtdForm(type: AbstractType<any>): jtd.JtdForm {
+  const typeName = type.getTypeName();
```

Comment:
Make this method work `Schema` instead, which is returned from `AbstractType.getSchema()`.

```suggestion
export function toJtdForm(schema: Schema): jtd.JtdForm {
  const typeName = schema.kind;
```